### ISSUE TITLE
Cant get offline messages.

### ISSUE THREAD

**Issue by parthpatel1105** [robbiehanson/XMPPFramework#520]
I am trying to get offline message but i can't do this.
When app was terminate and after start app and join group for group chat at that time i could not get delay or offline messages.
I had made setting from openfire for receive this kind of messages but at that time server give all group messages.

Please help me. I am stuck at this stage.

**Comment by SushilSharma12** [comment]
Try to import XMPPMessage+XEP_0184.h in app delegate.  I was facing same issue in peer-peer chat. check here http://stackoverflow.com/questions/28635816/receive-offline-message-with-xmpp

**Comment by parthpatel1105** [comment]
Thanks for help.

I have to send message in group. So how can i know which member of that
group are online or not?

Please let me know as soon as possible.

On Wed, Apr 15, 2015 at 1:08 PM, SushilSharma12 <notifications@github.com>
wrote:

> Try to import XMPPMessage+XEP_0184.h in app delegate. I was facing same
> issue in peer-peer chat. check here
> http://stackoverflow.com/questions/28635816/receive-offline-message-with-xmpp
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robbiehanson/XMPPFramework/issues/520#issuecomment-93241951>
> .
>



-- 

Thanks,
Best Regards,
*Parth Patel*
+91 9925053715,

**Comment by SushilSharma12** [comment]
I didn't work with group chat yet. But, I guess it should be same like getting presence of user in peer-peer chat.
Send presence while entering a chat room NSXMLElement *presence = [NSXMLElement elementWithName:@"presence"];

Then, check for the online users in chat room like this 
- (void)xmppStream:(XMPPStream *)sender didReceivePresence:(XMPPPresence *)presence {
    
    NSString *presenceType = [presence type]; // online/offline
    NSString *myUsername = [[sender myJID] user];
    NSString *presenceFromUser = [[presence from] user];
    NSMutableDictionary *mutableDict;
    if (![presenceFromUser isEqualToString:myUsername]) {
        
        if ([presenceType isEqualToString:@"available"]) {
        
        }
        else if ([presenceType isEqualToString:@"unavailable"]) {

        }
    }
}
Are you now able to get the offline messages ?

**Comment by parthpatel1105** [comment]
No i could not get offline messages. If i am set *Don't Show History from
open fire than i could not get offline and online messages. Had u some idea
about this.*

I had attached image for open fire.[image: Inline image 1]

On Wed, Apr 15, 2015 at 5:18 PM, SushilSharma12 <notifications@github.com>
wrote:

> I didn't work with group chat yet. But, I guess it should be same like
> getting presence of user in peer-peer chat.
> Send presence while entering a chat room NSXMLElement *presence =
> [NSXMLElement elementWithName:@"presence"];
>
> Then, check for the online users in chat room like this
>
>    -
>
>    (void)xmppStream:(XMPPStream *)sender didReceivePresence:(XMPPPresence
>    *)presence {
>
>    NSString *presenceType = [presence type]; // online/offline
>    NSString *myUsername = [[sender myJID] user];
>    NSString *presenceFromUser = [[presence from] user];
>    NSMutableDictionary *mutableDict;
>    if (![presenceFromUser isEqualToString:myUsername]) {
>
>    if ([presenceType isEqualToString:@"available"]) {
>
>    }
>    else if ([presenceType isEqualToString:@"unavailable"]) {
>
>    }
>
>    }
>    }
>    Are you now able to get the offline messages ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robbiehanson/XMPPFramework/issues/520#issuecomment-93347497>
> .
>



-- 

Thanks,
Best Regards,
*Parth Patel*
+91 9925053715,

**Comment by nsc2781** [comment]
xep45

发自我的 iPhone

**Comment by parthpatel1105** [comment]
How can i use unique date and time for group message in iOS and android
both?
I have to send message between iOS and android. But sometime android device
time is less then iOS or vice versa. So how can i use unique time for send
and receive message in Group Chat??

Please send me replay as soon as possible.

On Wed, Apr 15, 2015 at 8:41 PM, nsc2781 <notifications@github.com> wrote:

> xep45
>
> 发自我的 iPhone
>
> > 在 2015年4月15日，19:07，parthpatel1105 <notifications@github.com> 写道：
> >
> > Thanks for help.
> >
> > I have to send message in group. So how can i know which member of that
> > group are online or not?
> >
> > Please let me know as soon as possible.
> >
> > On Wed, Apr 15, 2015 at 1:08 PM, SushilSharma12 <
> notifications@github.com>
> > wrote:
> >
> > > Try to import XMPPMessage+XEP_0184.h in app delegate. I was facing same
> > > issue in peer-peer chat. check here
> > >
> http://stackoverflow.com/questions/28635816/receive-offline-message-with-xmpp
> > >
> > > ―
> > > Reply to this email directly or view it on GitHub
> > > <
> https://github.com/robbiehanson/XMPPFramework/issues/520#issuecomment-93241951
> >
> > > .
> > >
> >
> >
> >
> > --
> >
> > Thanks,
> > Best Regards,
> > *Parth Patel*
> > +91 9925053715,
>
> > ―
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robbiehanson/XMPPFramework/issues/520#issuecomment-93439454>
> .
>



-- 

Thanks,
Best Regards,
*Parth Patel*
+91 9925053715,

**Comment by parthpatel1105** [comment]
How can i use unique date and time for group message in iOS and android both?
I have to send message between iOS and android. But sometime android device time is less then iOS or vice versa. So how can i use unique time for send and receive message in Group Chat??

Please send me replay as soon as possible.

**Comment by SushilSharma12** [comment]
Hello parth, sorry I wasn't able to solve your issue. I too have an issue with XMPP , can you please check it here ? 
https://github.com/robbiehanson/XMPPFramework/issues/524

**Comment by parthpatel1105** [comment]
I had found one option for send offline messages. I had uploaded message on
server using php web service and send it when user become online. SO this
issue was solved. But now i am stuck in unique message send time for
android and iOS. So can you help me in this problem?

On Tue, Apr 21, 2015 at 12:10 PM, SushilSharma12 <notifications@github.com>
wrote:

> Hello parth, sorry I wasn't able to solve your issue. I too have an issue
> with XMPP , can you please check it here ?
> #524 <https://github.com/robbiehanson/XMPPFramework/issues/524>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robbiehanson/XMPPFramework/issues/520#issuecomment-94654480>
> .
>



-- 

Thanks,
Best Regards,
*Parth Patel*
+91 9925053715,

**Comment by vishalkohad121** [comment]
Hi i am also not getting offline group chat message.How can i get group chat history using php code or this history stored in openfire database mysql.

**Comment by parthpatel1105** [comment]
XMPP Provide Plugins for Offline Messages. You can manage it in open fire. And you have to create plugin by using any PHP set up.

**Comment by vishalkohad121** [comment]
yes i have installed monitoring service plugin adn enabled the archiving setting also but it will only store offline message of one to one chat in ofoffline table of openfire mysql Database.But it is not storing offline messages of group chat in any of the table is there any way to get group chat messages of offline users using php code or any other plugin which i miss ?

**Comment by varunjoshi12** [comment]
Hi i am also not getting offline group chat message.How can i get group chat history using php code or this history stored in openfire database mysql.And yes i have also installed monitoring service plugin and enabled the archiving setting also but it will only store offline message of one to one chat in ofoffline table of openfire mysql Database.But it is not storing offline messages of group chat in any of the table is there any way to get group chat messages of offline users using php code or any other plugin which i miss ?

**Comment by IOSDeveloperProgrammer** [comment]
@parthpatel1105  @varunjoshi12 @SushilSharma12  Hi, How can i get offline messages ? Anyone solve this issue. I'm stuck on this issue, Please help me in this problem.

**Comment by atmohsin** [comment]
HI Varunjoshi12

We also face the same issue. Did you manage to get any solutions to this problem?  Please share the solutions.

Thanks
Mohsin

**Comment by atmohsin** [comment]
Hi, 

Is this issue is with this framework or XMPP server (Openfire) issue?

Thanks
Mohsin
